src/controllers/eventController.js

```diff
@@ -0,0 +1,249 @@
+          userId: userId,
+          rsoId: null
+        }
+      }),
+      db.Membership.count({
+        where: {
+          universityId: universityId,
+          userId: userId,
+          rsoId: rsoId
+        }
+      })
+    )
+
+    let allowedPrivacy = eventPrivacyLevels.PUBLIC //public
+    Promise.all(membershipCount)
+      .then((result) => {
+        if (result[0] > 0) {
+          allowedPrivacy = eventPrivacyLevels.PRIVATE
+        }
+        if (result[1] > 0) {
+          allowedPrivacy = eventPrivacyLevels.RSO
+        }
+        resolve(allowedPrivacy)
+      })
+  })
+}
+
+class EventController extends ApiController {
+  index (req, res, next) {
+    //Make sure that the user is logged in.
+    if (!req.isAuthenticated()) {
+      return next(new ApiError.UserNotAuthenticated())
+    }
+
+    //Get the required parameters
+    let params = _.merge(
+      helpers.requireParams([
+        'name',
+        'description',
+        'startTime',
+        'endTime',
+        'privacy',
+        'category',
+        'contactPhone',
+        'contactEmail',
+        'createdById'
+      ], req.body, true),
+      helpers.requireParams([
+        'id',
+        'universityId',
+        'rsoId',
+      ], req.params),
+    )
+
+    let allowedPrivacy = eventPrivacyLevels.PUBLIC
+    getAllowedPrivacy(params.universityId, req.user.id, params.rsoId)
+      .then((privacy) => {
+        allowedPrivacy = privacy
+        let payload = params
+        db.Event.findAll({
+          where: payload
+        }).then((events) => {
+          res.json(_.filter(events, (evt) => {
+            return evt.privacy >= allowedPrivacy
+          }))
+        })
+      })
+  }
+
+  show (req, res, next) {
+    //Make sure that the user is logged in
+    if (!req.isAuthenticated()) {
+      return next(new ApiError.UserNotAuthenticated())
+    }
+
+    // Get the required parameters
+    let params = helpers.requireParams([
+      'universityId',
+      'rsoId',
+      'id'
+    ], req.params)
+
+    getAllowedPrivacy(params.universityId, req.user.id, params.rsoId)
+      .then((privacy) => {
+        let payload = params
+        db.Event.fineOne({
+          where: payload
+        }).then((rso) => {
+          if(privacy.rso >= privacy){
+            res.json(rso)
+          } else {
+            return next(new ApiError.EventPrivacyRestriction())
+          }
+        }, (response) => {
+          console.log('Failed to locate event in RSO', response)
+          return next(new ApiError.NoEventinRso({ action : 'event#show', params: params, response: response }))
+        })
+      })
+  }
+
+  create (req, res, next) {
+    // Make sure that the user is logged in.
+    if (!req.isAuthenticated()) {
+      return next(new ApiError.UserNotAuthenticated())
+    }
+
+    //make sure user is member of rso
+    if(!Membership.count({ where: { userId: userId, rsoId: rsoId, inactiveAt: null } })) {
+      return next(new ApiError.UserNotInRso())
+    }
+
```

Comment:
Yeah it would, but while this promise is resolving, the rest of the code below it will be running; you only want it to run if the condition in this promise is met. So, you should wrap the rest of the code below this promise in another function and it call it respectively.

```
Membership.count({ where: { userId: userId, rsoId: rsoId, inactiveAt: null } }) {
  .then((count) => {
    if (!count) {
      next(new ApiError.UserNotInRso())
    } else {
        performCreate();
    }
  }) 
}

let performCreate = () => {
    // rest of the function
}
```